src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -382,13 +382,21 @@ static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
 *
 * emptyAttributeValue -
 *
-* Check that the attribute doesn't have any value, taking into account its multi-valuated
-* nature
+* Check that the attribute doesn't have any value, taking into account its multi-valuated nature.
+*
+* NOTE
+*   1. ValueTypeNumber and ValueTypeBoolean are always non-empty.
+*   2. In API version 2, the value of a ContextAttribute CANNOT be left out
```

Comment:
{
        "type": "foo"
    }

As "value" is not present inside this object, it is considered a compound value and thus, the attribute has a value.
As I said, there is no was in v2 to create an attribute without value